src/Models/AbstractEntity.php

```diff
@@ -778,14 +778,18 @@ public function update(ContentParamsInterface $params): bool
             $content = $this->readOne()['body'] . $content;
         }
         // ensure no changes happen on entries with immutable permissions
+        // admins can override the immutability of an entity's permissions. See #5800
         if ($params->getTarget() === 'canread' || $params->getTarget() === 'canwrite') {
-            if (($this->entityData[$params->getTarget() . '_is_immutable'] ?? 0) === 1 && (!($this instanceof AbstractTemplateEntity))) {
-                throw new ImproperActionException(_('Cannot modify permissions on entry with immutable permissions.'));
+            if (($this->entityData[$params->getTarget() . '_is_immutable'] ?? 0) === 1
+                && !($this instanceof AbstractTemplateEntity)
+                && !($this->Users->isAdmin)
```

Comment:
I was thinking that the check should be "user is Admin in the team in which the concrete entity is associated" rather than "user is Admin". But then if you don't have write access you cannot edit the permissions anyway so it seems enough.